cdap-common/src/main/java/co/cask/cdap/common/internal/guava/ClassPath.java

```diff
@@ -428,12 +444,13 @@ private void scanJar(File file, ClassLoader classloader) throws IOException {
           scan(uri, classloader);
         }
         Enumeration<JarEntry> entries = jarFile.entries();
+        URL baseURL = file.toURI().toURL();
         while (entries.hasMoreElements()) {
           JarEntry entry = entries.nextElement();
           if (entry.isDirectory() || entry.getName().equals(JarFile.MANIFEST_NAME)) {
             continue;
           }
-          resources.add(ResourceInfo.of(entry.getName(), classloader));
+          resources.add(ResourceInfo.of(baseURL, entry.getName(), classloader));
         }
       } finally {
```

Comment:
although unrelated to this PR, should we use resources syntax for ``jarFile`` here?